question-service/service/questionService.js

```diff
@@ -0,0 +1,19 @@
+import { getNumQuestions, getQuestionById } from "../repository/questionRepository.js";
+
+export const getQuestionService = async () => {
+	let rand = 0;
+	while (rand === 0) {
+		rand = Math.ceil(Math.random() * await getNumQuestions());
+	}
```

Comment:
I think this function can be abstracted out, since it is repeated twice. Perhaps can put in a util file, but allow it to take in the blacklist. This is to ensure DRY principle